www/topstories.php

```diff
@@ -17,10 +17,19 @@
 $current_page = get_current_page();
 $offset=($current_page-1)*$page_size;
 
+$format = mb_strtolower($_GET['format']);
+$date = $_GET['date'];
+// default date for daily format = today
+if ( $format == 'daily' && empty($date)) {
+    $date = date('Y-m-d');
+}
+
 // Select a month and year
 if (!empty($_GET['month']) && !empty($_GET['year']) && ($month = (int) $_GET['month']) > 0 && ($year = (int) $_GET['year'])) {
 	$sql = "SELECT SQL_CACHE link_id, link_votes as votes FROM links, sub_statuses WHERE id = ".SitesMgr::my_id()." AND YEAR(date) = $year AND MONTH(date) = $month AND status = 'published' AND link = link_id ORDER BY link_votes DESC ";
 	$time_link = "YEAR(date) = $year AND MONTH(date) = $month AND";
+} else if (!empty($date)) {
+    $sql = "SELECT SQL_CACHE link_id, link_votes as votes FROM links, sub_statuses WHERE id = ".SitesMgr::my_id()." AND DATE(date) = '$date' AND status = 'published' AND link = link_id ORDER BY link_votes DESC ";
```

Comment:
No se está validando el parámetro `date`, así que es susceptible de SQL injection.

Por ejemplo, si envías el siguiente valor `' OR 2 != '1` la query resultante incluiría la siguiente cláusula WHERE:

`[...] AND DATE(date) = '' OR 2 != '1' AND status = 'published' [...]`

De primeras lo único que sucedería es que no se aplicaría ningún filtro por fecha, pero se podría aprovechar para hacer JOINs con otras tablas y adivinar así correos electrónicos, emails o ips.